sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/GcpApiSurfaceTest.java

```diff
@@ -73,6 +73,8 @@ public void testGcpApiSurface() throws Exception {
             Matchers.equalTo(com.google.api.gax.rpc.ApiException.class),
             Matchers.<Class<?>>equalTo(com.google.api.gax.rpc.StatusCode.class),
             Matchers.<Class<?>>equalTo(com.google.common.base.Function.class),
+            Matchers.<Class<?>>equalTo(com.google.common.base.Optional.class),
+            Matchers.<Class<?>>equalTo(com.google.common.base.Supplier.class),
```

Comment:
Copy from the commit message:

GcpApiSurfaceTest failed after upgrading bigtable-client-core to 1.19.1
because its CallOptionsConfig uses Guava's Optional as one of the
return values of APIs.
(Guava's Optional has its Supplier in one of API argument.)
googleapis/java-bigtable-hbase#2717

Note that, while these classes have counterparts in Java 8,
bigtable-client-core targets Java 7 as minimum required Java version.

This commit adds the two classes into the allow-list in GcpApiSurfaceTest.